### PR TITLE
Added Error line/column info.

### LINKED ISSUE #307: Getting position/span from Error
handlebars is using the line/col number from pest `Error` for error reporting. From 2.0, it seems no way to get from the `Error` struct. I suggest to hold the `Position` or `Span` within `InputLocation` and provide getters for them.

### PR DESCRIPTION
**Pull request by dragostis**
Fixes #307. @sunng87, is this a fine solution?

### CODE REVIEW

File: pest/src/error.rs [hunk 1]
```diff
@@ -230,11 +237,17 @@ impl<R: RuleType> Error<R> {
...
+    }
+
     fn spacing(&self) -> String {
-        let line = if let Some((line, _)) = self.end {
-            cmp::max(self.start.0, line)
-        } else {
-            self.start.0
+        let line = match self.line_col {
+            LineColLocation::Pos((line, _)) => line,
+            LineColLocation::Span(_, (line, _)) => cmp::max(self.start().0, line)
```
Comment (CAD97) [inline pest/src/error.rs]: Wouldn't this be better written as

```rust
LineColLocation::Span((head, _), (tail, _)) => cmp::max(head, tail)
```

Comment (dragostis) [reply pest/src/error.rs]: Definitely. It slipped through. Feel free to push the fix and merge if you have the time. I'm going to be travelling until Wednesday.

Comment (dragostis) [reply pest/src/error.rs]: Probably call them `start_line`, `end_line` though?

**Comment by sunng87** [comment]
Thanks! This looks good for my requirement.

It just seems a little verbose to have both `InputLocation` and `LineColLocation` under `Error`. Semantically I think the function of `LineColLocation` should belong to `InputLocation`.

**Comment by dragostis** [comment]
@sunng87, how would you define the `InputLocation` data structure?

**Comment by sunng87** [comment]
Perhaps we can have a light-weighted `Position` struct that stores `pos`, `line_no` and `col_no`. `InputLocation` is still an enum of `Pos` and `Span`, but each holding the light weight `Position` so we can get enough information . This breaks API compatibility though. 

Anyway, this patch fixes it well while maintained backward compatibility and I'm OK with it overall. We can refactor it during next major release.

**Comment by dragostis** [comment]
bors r=CAD97

**Comment by sunng87** [comment]
Cool! Could you please create a release for this?

**Comment by dragostis** [comment]
Done. :)

**Pull request merged**